dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelAIPluginExtensions.cs

```diff
@@ -0,0 +1,439 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+public static class KernelAIPluginExtensions
+{
+    /// <summary>
+    /// Imports an AI plugin that is exposed as an OpenAPI v3 endpoint or through OpenAI's ChatGPT format.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="filePath">The file path to the AI Plugin</param>
+    /// <param name="executionParameters">Skill execution parameters.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportAIPluginAsync(
+        this IKernel kernel,
+        string skillName,
+        string filePath,
+        OpenApiSkillExecutionParameters? executionParameters = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(kernel);
+        Verify.ValidSkillName(skillName);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+        var internalHttpClient = HttpClientProvider.GetHttpClient(kernel.Config, executionParameters?.HttpClient, kernel.Log);
+#pragma warning restore CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+
+        var pluginJson = await LoadJsonFromFilePath(
+            kernel,
+            filePath,
+            executionParameters,
+            internalHttpClient,
+            cancellationToken).ConfigureAwait(false);
+
+        if (TryParseAIPluginForUrl(pluginJson, out var openApiUrl))
+        {
+            return await kernel
+                .ImportAIPluginAsync(
+                    skillName,
+                    new Uri(openApiUrl),
+                    executionParameters,
+                    cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        //assume OpenAPI v3
+        var parser = new OpenApiDocumentParser(kernel.Log);
+
+        using (var documentStream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(pluginJson)))
+        {
+            var operations = await parser.ParseAsync(documentStream, executionParameters?.IgnoreNonCompliantErrors ?? true, cancellationToken).ConfigureAwait(false);
```

Comment:
Allowing the ignoring of non-compliant errors by default will prevent SK consumer code developers from being aware of the non-compliant OpenAPI plugin they may want to work with. As a result, some issues may pop up later during the plugin's execution phase rather than the import phase, when it may be a little bit tricky to debug without knowing that the plugin does not follow the OpenAPI specification. 

Disabling it by default, on the other hand, will allow SK consuming code developers to make an explicit decision whether to use a particular plugin or not after a failure during the import phase, it will also provide an additional context that will make troubleshooting easier if the plugin misbehaves during the execution phase.

Reply:
Ah good catch, I originally considered flipping this then forgot to unwind. Switched back to false.